packages/squiggle-lang/__tests__/Distributions/Invariants/Means_test.res

```diff
@@ -53,88 +30,115 @@ describe("Mean", () => {
-    ) => {
-    let dist1 = dist1'->E.R2.fmap(x=>DistributionTypes.Symbolic(x))->E.R2.fmap2(s=>DistributionTypes.Other(s))
-    let dist2 = dist2'->E.R2.fmap(x=>DistributionTypes.Symbolic(x))->E.R2.fmap2(s=>DistributionTypes.Other(s))
+    distOp: (
+      DistributionTypes.genericDist,
+      DistributionTypes.genericDist,
+    ) => result<DistributionTypes.genericDist, DistributionTypes.error>,
+    description: string,
+    floatOp: (float, float) => float,
+    dist1': SymbolicDistTypes.symbolicDist,
+    dist2': SymbolicDistTypes.symbolicDist,
+    ~epsilon: float,
+  ) => {
+    let dist1 = dist1'->DistributionTypes.Symbolic // ->DistributionTypes.Other
+    let dist2 = dist2'->DistributionTypes.Symbolic // ->DistributionTypes.Other
     let received =
-      E.R.liftJoin2(distOp, dist1, dist2)
-      ->E.R2.fmap(mean)
-      ->E.R2.fmap(run)
-      ->E.R2.fmap(toFloat)
+      distOp(dist1, dist2)->E.R2.fmap(mean)->E.R2.fmap(run)->E.R2.fmap(toFloat)->E.R.toExn
     let expected = floatOp(runMean(dist1), runMean(dist2))
     switch received {
-    | Error(err) => impossiblePath(description)
-    | Ok(x) =>
-      switch x {
-      | None => impossiblePath(description)
-      | Some(x) => x->expect->toBeSoCloseTo(expected, ~digits)
-      }
+    | None => expectImpossiblePath(description)
+    | Some(x) => expectErrorToBeBounded(x, expected, ~epsilon)
     }
   }
+}
 
-  describe("addition", () => {
-    let testAdditionMean = testOperationMean(algebraicAdd, "algebraicAdd", \"+.")  
+let {
+  algebraicAdd,
+  algebraicMultiply,
+  algebraicDivide,
+  algebraicSubtract,
+  algebraicLogarithm,
+  algebraicPower,
+} = module(DistributionOperation.Constructors)
 
-    testAll("homogeneous addition", zipDistsDists, dists => {
-      let (dist1, dist2) = dists
-      testAdditionMean(dist1, dist2)
+let algebraicAdd = algebraicAdd(~env)
+let algebraicMultiply = algebraicMultiply(~env)
+let algebraicDivide = algebraicDivide(~env)
+let algebraicSubtract = algebraicSubtract(~env)
+let algebraicLogarithm = algebraicLogarithm(~env)
+let algebraicPower = algebraicPower(~env)
+
+let {testOperationMean, distributions, pairsOfDifferentDistributions, epsilon} = module(Internals)
+
+describe("Means invariant", () => {
+  describe("for addition", () => {
+    let testAdditionMean = testOperationMean(algebraicAdd, "algebraicAdd", \"+.", ~epsilon)
+
+    testAll("of two of the same distribution", distributions, dist => {
+      E.R.liftM2(testAdditionMean, dist, dist)->E.R.toExn
     })
 
-    testAll("heterogeneous addition (1)", combinations, dists => {
+    testAll("of two different distributions", pairsOfDifferentDistributions, dists => {
       let (dist1, dist2) = dists
-      testAdditionMean(dist1, dist2)
+      E.R.liftM2(testAdditionMean, dist1, dist2)->E.R.toExn
     })
 
-    testAll("heterogeneous addition (commuted of 1 (or; 2))", combinations, dists => {
+    testAll("of two difference distributions", pairsOfDifferentDistributions, dists => {
```

Comment:
different*

Reply:
good catch, will be in next push.

Reply:
maybe, "of two different distributions with swapped ordering"